src/rust/aggregator/src/data/user_history.rs

```diff
@@ -300,14 +323,16 @@ impl Data for UserHistory {
                     }
                 }
                 (Some(fill), None) => {
-                    aggregate_fill(
-                        &mut transaction,
-                        &fill.size,
-                        &fill.maker_order_id,
-                        &fill.market_id,
-                        &fill.time,
-                    )
-                    .await?;
+                    if fill.maker_address == fill.emit_address {
```

Comment:
This check should go in `aggregate_fill`, because now market orders and swaps will not have their order size or status updated on fill

Reply:
Actually, deduping here works, but there needs to be extra logic in `aggregate_fill` that updates the maker and the taker side of the order